packages/api/endpoints/rules.js

```diff
@@ -221,34 +220,16 @@ async function del(req, res) {
     }
   }
 
-  try {
-    // Save DynamoDB rule to recreate record in case of deletion failure
-    apiRule = await ruleModel.get({ name });
-  } catch (error) {
-    if (!(error instanceof RecordDoesNotExist)) {
-      throw error;
-    }
-  }
-
-  try {
-    await createRejectableTransaction(knex, async (trx) => {
-      await rulePgModel.delete(trx, { name });
-      await ruleModel.delete(apiRule);
-      await deleteRule({
-        esClient,
-        name,
-        index: process.env.ES_INDEX,
-        ignore: [404],
-      });
+  await createRejectableTransaction(knex, async (trx) => {
+    await rulePgModel.delete(trx, { name });
+    await deleteRule({
+      esClient,
+      name,
+      index: process.env.ES_INDEX,
+      ignore: [404],
     });
-  } catch (error) {
-    // Delete is idempotent, so there may not be a DynamoDB
-    // record to recreate
-    if (apiRule) {
-      await ruleModel.create(apiRule);
-    }
-    throw error;
-  }
+    if (rule) await deleteRuleResources(knex, rule);
```

Comment:
This is necessary in the case that the record exists in Elasticsearch but not PG. Though, in the case that the rule record only exists in Elasticsearch, can we still retrieve values we need to delete rule resources (name, log_event_arn, arn, etc) ?